cluster-autoscaler/utils/drain/drain.go

```diff
@@ -73,18 +71,19 @@ func GetPodsForDeletionOnNodeDrain(
 		daemonsetPod := false
 		replicated := false
 
-		sr, err := CreatorRef(pod)
-		if err != nil {
-			return []*apiv1.Pod{}, fmt.Errorf("failed to obtain refkind: %v", err)
-		}
+		controllerRef := ControllerRef(pod)
 		refKind := ""
-		if sr != nil {
-			refKind = sr.Reference.Kind
+		if controllerRef != nil {
+			refKind = controllerRef.Kind
 		}
 
+		// For now, owner controller must be in the same namespace as the pod
```

Comment:
Is there some plan to change this (add namespace to ref)? If so it may be worth creating an issue to keep track of this and remind us to fix this later on.

Reply:
Not sure about this, comment is based on https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L276

Reply:
Ok, looks like it's not likely to change anytime soon, so just ignore my comment above.